### PR TITLE
HackerSM64 2.0.3 Release: Audio Crash

### PR DESCRIPTION
**Pull request by gheskett**
We believe this crash to most likely be a result of invalid reads from the dcache.

An example of invalid data retrieval was tracked as far as this line: `audio_dma_copy_immediate((uintptr_t) data, gAlCtlHeader, 0x10);` The data that was written to `gAlCtlHeader`, when read, was incorrect. However, after reading from the same address (+ 0x20000000 to hopefully bypass the cache), the results _sometimes_ read correctly from the exact same variable and the line after the initial read. This is rather weird to observe but it's strong indication of an issue revolving around the dcache.

The solution, after some reading, is simply to make sure every buffer being written to via DMA is 16-byte aligned in memory. DMAs should have the virtual addresses be 8-byte aligned already, but 16-byte alignment of both address and size should pretty much guarantee the data doesn't share any cache lines with other variables that could corrupt them later.

This solution here is not technically confirmed to fix the issue for sure, but there's plenty of evidence to back it up based on my observations and the overall uncanny behavior of this issue.

### CODE REVIEW

**Review by Reonu — APPROVED** [review]

**Review by thecozies — APPROVED** [review]
# sick

File: src/audio/heap.h [hunk 1]
```diff
@@ -64,9 +70,14 @@ struct SoundMultiPool {
     /*     */ u32 pad2[4];
 }; // size = 0x1D0
 
+#ifdef VERSION_SH
```
Comment (arthurtilly) [inline src/audio/heap.h:73]: why is this ifdefed to shindou?

Comment (gheskett) [reply src/audio/heap.h:73]: It's unused in the other versions, and primarily it just means darkening their displays in VSCode

Comment (arthurtilly) [reply src/audio/heap.h:73]: fair, looks good then

**Pull request merged**